jupyterhub_moss/spawner.py

```diff
@@ -235,8 +244,11 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
         ):
             options["exclusive"] = True
 
-        # Specific handling of jupyterlab
-        self.default_url = "/lab" if options.get("jupyterlab", False) else ""
+        # Specific handling of landing URL (e.g., to start jupyterlab)
+        self.default_url = options.get("default_url", "")
+
+        if "root_dir" in options:
+            self.notebook_dir = options["root_dir"]
```

Comment:
`root_dir` is the new name, `notebook_dir` is deprecated.

Reply:
Is it working for both JupyterLab 2 and 3 ?

Reply:
Yes, I checked, it works with both.
Here we set `jupyterhub` not JupyterLab directly.